packages/client/src/App.tsx

```diff
@@ -17,28 +17,48 @@ import useConnectAuthToStore from "@/react-redux-firebase-auth/hooks/useConnectA
 import { currentTheme } from "@/themes";
 import { NotificationsProvider } from "./features/notifications/context";
 
+import { closeAllModals } from "./features/modal/actions";
+
 const App: React.FC = () => {
   // connect auth to store to recieve firebase SDK's auth updates
   // through redux store
   useConnectAuthToStore(getAuth(), store);
+  const history = useHistory();
+
+  // Subscribe to history changes and dispatch close modal action
+  // to close any open modal when navigating away from the page
+  React.useEffect(() => {
+    const unlisten = history.listen(() => {
+      store.dispatch(closeAllModals);
+    });
+    return () => unlisten();
+  }, [history]);
 
   return (
-    <BrowserRouter>
-      <ReduxProvider store={store}>
-        <StyledEngineProvider injectFirst>
-          <ThemeProvider theme={currentTheme}>
-            <NotificationsProvider
-              timeouts={{ minTimeout: 1200, maxTimeout: 2000 }}
-            >
-              <CssBaseline />
-              <AppContent />
-              <Modal />
-            </NotificationsProvider>
-          </ThemeProvider>
-        </StyledEngineProvider>
-      </ReduxProvider>
-    </BrowserRouter>
+    <ReduxProvider store={store}>
+      <StyledEngineProvider injectFirst>
+        <ThemeProvider theme={currentTheme}>
+          <NotificationsProvider
+            timeouts={{ minTimeout: 1200, maxTimeout: 2000 }}
+          >
+            <CssBaseline />
+            <AppContent />
+            <Modal />
+          </NotificationsProvider>
+        </ThemeProvider>
+      </StyledEngineProvider>
+    </ReduxProvider>
   );
 };
 
-export default App;
+/**
+ * An additional wrapper component, wrapping the app in a browser
+ * router to make the router context available at top level.
+ */
+const RouterWrapped = () => (
```

Comment:
Why wrap App separately, as opposed to wrap it in place (how it used to be)?

Reply:
In order to be able to ‘useHistory’ in App the App component itself needs to have access to the router context and wrapping it in <Router> (like here), we create an instance of router context which is then accessible in the App “component”